modules/hitl2/src/views/full/Components/UserProfile.tsx

```diff
@@ -25,56 +23,48 @@ const UserProfile: FC<Props> = props => {
     setUser(_.get(JSON.parse(props.conversation.event), 'state.user', {}))
   }, [props.conversation])
 
-  function defaultUserName(): string {
-    const key = hash(_.pick(props.conversation, ['channel', 'threadId']))
-
-    if (!userNames[key]) {
-      userNames[key] = haiku.haikunate({ delimiter: ' ' })
-      setUserNames(userNames)
-    }
-
-    return userNames[key]
-  }
-
-  function variables() {
-    return _.omit(user, 'fullname', 'email')
+  if (!defaultUserName.current) {
```

Comment:
Clever, but I don't think this handles assigning the same name to a user that creates multiple escalations. That's what `key = hash(_.pick(props.conversation, ['channel', 'threadId']))` was for. You could also use the `escalation.target` property now that we have added it (see recent changes in #4021)

Reply:
I don't know why, but I completely missed it, can't remember if I forgot to put it back or if I didn't see it at all 😂